lib/ansible/plugins/lookup/mongodb.py

```diff
@@ -0,0 +1,167 @@
+# (c) 2016, Marcos Diez <marcos@unitron.com.br>
+# https://github.com/marcosdiez/
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import (absolute_import, division, print_function)
+from __future__ import unicode_literals
+from ansible.module_utils.six import string_types
+import datetime
+
+__metaclass__ = type
+
+try:
+    from pymongo  import ASCENDING, DESCENDING
+    from pymongo.errors import ConnectionFailure
+    from pymongo import MongoClient
+except ImportError:
+    try:  # for older PyMongo 2.2
+        from pymongo import Connection as MongoClient
+    except ImportError:
+        pymongo_found = False
+    else:
+        pymongo_found = True
+else:
+    pymongo_found = True
+
+
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+
+class LookupModule(LookupBase):
+
+    def _fix_sort_parameter(self, sort_parameter):
+        if sort_parameter is None:
+            return sort_parameter
+
+        if not isinstance(sort_parameter, list):
+            raise AnsibleError("Error. Sort parameters must be a list, not [ {} ]".format(sort_parameter))
+
+        for item in sort_parameter:
+            self._convert_sort_string_to_constant(item)
+
+        return sort_parameter
+
+    def _convert_sort_string_to_constant(self, item):
+        original_sort_order = item[1]
+        sort_order = original_sort_order.upper()
+        if sort_order == "ASCENDING":
+            item[1] = ASCENDING
+        elif sort_order == "DESCENDING":
+            item[1] = DESCENDING
+        #else the user knows what s/he is doing and we won't predict. PyMongo will return an error if necessary
+
+    def convert_mongo_result_to_valid_json(self, result):
+        if result is None:
+            return result
+        if isinstance(result, (int, long, float, bool)):
+            return result
+        if isinstance(result, string_types):
+            return result
+        elif isinstance(result, list):
+            new_list = []
+            for elem in result:
+                new_list.append(self.convert_mongo_result_to_valid_json(elem))
+            return new_list
+        elif isinstance(result, dict):
+            new_dict = {}
+            for key in result.keys():
+                value = reslut[key] # python2 and 3 compatible....
```

Comment:
@jbence would you be kind enough to code review the PR that fix this issue ?

AFAIK I need two code reviewers to get the PR merged and the whole module is useless without it.

https://github.com/ansible/ansible/pull/20478

Thanks!

@Dufgui @Lujeni  you are also summoned. it's a one line!